test/test_browser.py

```diff
@@ -4795,28 +4795,19 @@ def test_browser_run_from_different_directory(self):
 
   # Similar to `test_browser_run_from_different_directory`, but asynchronous because of `-sMODULARIZE`
   def test_browser_run_from_different_directory_async(self):
-    for args, creations in [
-      (['-sMODULARIZE'], [
-        'Module();',    # documented way for using modularize
-        'new Module();' # not documented as working, but we support it
```

Comment:
This is the other odd behavior which happened with work by accident before but not after.    Basically you can call `new Foo()` with a promise-returning function but not with an explicitly `async` function.

Reply:
Makes sense. But please

1. Add this breaking change to the changelog, and
2. Add a clear runtime error in ASSERTIONS mode, with a test.

With those, I think the risk is mitigated.

Reply:
I updated the ChangeLog, but I don't think there is any way to detect or allow `new Module()` if `Module` is `async` so I'm not sure there is anything can do in `ASSERTIONS` builds.

One possible option is to add another wrapper than is not async:

e.g. 

```
function Module() {
   async function InnerModule() {
       ...
   }
   return InnterModule();
}
```

Do you think its worth doing this?

Reply:
I think in an ASSERTIONS build that would be useful to do, yeah. A little ugly but if it saves users time, I think it's worth it.

Reply:
This landed in #23210